app/ui/browser/reducers/ui-state.js

```diff
@@ -23,10 +23,14 @@ export default function uiState(state = initialState, action) {
       return state.set('statusText', action.text);
 
     case types.LOCATION_CHANGED:
+      return state.setIn(['userTypedLocation', action.pageId], action.payload.text);
 
-      // Fallthrough.
     case types.SET_USER_TYPED_LOCATION:
-      return state.setIn(['userTypedLocation', action.pageId], action.payload.text);
+      return state.setIn(['userTypedLocation', action.pageId], action.payload.text)
```

Comment:
not a huge deal, but might want to use withMutations to avoid creating the intermediate object that is returned from setIn. See an [example here](https://github.com/mozilla/tofino/blob/c959b42c374f97356e06b28dcd65fe492d9a5efa/app/ui/browser/reducers/pages.js#L82)